src/analysis/sum_stat.py

```diff
@@ -0,0 +1,205 @@
+            df.drop_duplicates(subset=[ss_info[ss]['pos']]) \
+                .sort_values(by=[ss_info[ss]['pos']]) \
+                .reset_index(drop=True)
+
+        for pvalue in ss_info[ss]['p']:
+            if df_filtered[pvalue].max() > 1:
+                df_filtered[pvalue] = np.log10(df_filtered[pvalue])
+            if df_filtered[pvalue].min() < 0:
+                df_filtered[pvalue] = df_filtered[pvalue].apply(lambda x: x * -1)
+
+        keep_same = {ss_info[ss]['pos'], ss_info[ss]['id']}
+        df_filtered.columns = ['{}{}'.format(c, '' if c in keep_same else '_'+ss_info[ss]['tag']) for c in df_filtered.columns]
+        return df_filtered
+
+    def __merge_data(self, df_dict: dict, ss_info: dict):
+        first_time = True
+        for ss in df_dict:
+            if first_time:
+                previous_ss = df_dict[ss].rename(columns={ss_info[ss]['pos']: 'pos', ss_info[ss]['id']: 'id'})
+                first_time = False
+                continue
+            df = df_dict[ss].rename(columns={ss_info[ss]['pos']: 'pos', ss_info[ss]['id']: 'id'})
+            previous_ss = pd.merge(df, previous_ss, how="inner", on=["pos", "id"])
+
+        return previous_ss
+
+    def __normalise_pvalues(self, df: pandas.DataFrame, ss_info: dict):
+        for pvalue in ss_info['p']:
+            df[pvalue] = (df[pvalue] - df[pvalue].min()) / \
+            (df[pvalue].max() - df[pvalue].min())
+
+    def prepare_data(self, *ss_and_tag: tuple[str], working_dir: str, to_file='ss.tsv', norm=True) -> pandas.DataFrame:
+        '''
+        Returns aggregated and intersected summary statistics dataframe.
+        All summary statistics files must contain position column and at least one of them should have id column.
+        All necessary columns will be detected automatically with regex.
+
+                Parameters:
+                        ss_and_tag (tuple): Any number of tuples with format: (path_to_summary_stat.tsv, ancestry_tag)
+                        working_dir (str): Path to a dir to store all lists into
+                        to_file (str or None): Path to output dataframe, if None won't create one
+                        norm (bool): Perform min max normalisation for each p-value column separately
+
+                Returns:
+                        merged_df (pandas.DataFrame): Dataframe with position, id and all -log10(p-values) columns from all input files
+        '''
+        ss_info = {}
+        for i, ss_tag in enumerate(ss_and_tag):
+            ss, tag = ss_tag
+            with open(ss, 'r') as ss_file:
+                header = ss_file.readline().strip('\n').split('\t')
+
+            logging.info(f'Reading basic info about {ss}...')
+            ss_info[ss] = self.__get_ss_info(header, tag)
+
+            logging.info(f'Preparing list of positions for {ss}...')
+            pos_list_dist = os.path.join(working_dir, f'pos{i}.list')
+            subprocess.call("awk -F '\t' '{ print $%s }' %s | sort | uniq > %s"
```

Comment:
please provide a comment for each of the shell commands that briefly explains what they do

Reply:
Don't you think that  `logging.info(...)` describes them already?